### ISSUE TITLE
Still having issues with CloudFlare Proxy and NGINX SSL certs :(

### ISSUE THREAD

**Issue by SpikeyDan** [Ylianst/MeshCentral#2653]
Hi There!

I currently have my DNS in Cloudflare and utilise their free Proxy service to hide my WAN IP. I also set the SSL settings in Cloudflare to this: 

Full (strict)
Encrypts end-to-end, but requires a trusted CA or Cloudflare Origin CA certificate on the server

I am having issues adding new agents. They do not seem to check into meshcentral over the WAN. One example is my android phone being able to connect via LAN and when I am on my 4G it sticks at "authenticating". I have another Windows server on my LAN though that fails to check in. I have made sure the Windows Firewall was disabled for testing. 

Here is my certhashes output:

```
> certhashes
AgentCertHash: 489aa51895c2205c4b3823a661a1b1dbdbebfc4be8fd58d24adfb1e9b90de716bdbc3b21085693efbcdf2b89d829a6da
webCertificateHash (): 1939E2D56E44BFEC5EF607EEFEA1EFFACF15152CCBA728A25DB0C21B9CE71E4C50D3E4CE773DABAD9CC157646E50D6F5
webCertificateFullHash (): F28B3843C5E8B653C94C7337750981113A04D9ED3918020E4FD14DFB6272AFB99B498B003A5790DDCAEAB7BFBEC85F3D
defaultWebCertificateHash: 0E1C1E149FBC63FF6C26EC2816A2B01A82A7DC7EF0F7BEF0E28411673F50AC51CBBDD763C93572C53B0CD36441E3A793
defaultWebCertificateFullHash: 97EFDC1270FBF811F28681C5B542FD1BDCC651083FE7EFF36C11169ECE7653AEA13D5DCB5179E61A79B8811C9FC8AE50
```

Here is my config. I have tried so many different combinations of config changes and searched here and online to no avail.
I would really appreciate any help I can get on this one as I love the project and want to use it.

```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "cert": "mesh.mydomain.com",
    "WANonly": true,
    "_LANonly": true,
    "sessionKey": "password",
    "port": 4430,
    "aliasPort": 443,
    "redirPort": 800,
    "tlsoffload": "192.168.150.161",
    "trustedproxy": "CloudFlare",
    "redirAliasPort": 80
  },
  "domains": {
    "": {
      "certurl": "https://mesh.domain.com:443/",
      "title": "DansMesh",
      "_title2": "DansMesh",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "production": false
  }
}
```

192.168.150.161 is my NGINX server and MeshCentral is 192.168.150.160.

Thanks :)

**Comment by Ylianst** [comment]
Hi. So, you can't have both these lines at the same time:

```
    "tlsoffload": "192.168.150.161",
    "trustedproxy": "CloudFlare",
```

You ether don't do TLS and trust "192.168.150.161" or, you do TLS and trust "CloudFlare". I have a video on setting up [MeshCentral with NGINX](https://www.youtube.com/watch?v=YSmiLyKSX2I), that covers the basic theory. In general, your chaining two reverse proxies in a row and there is not sufficient information to debug this from your post. For example are you intending to do TLS between NGINX and MeshCentral?

**Comment by SpikeyDan** [comment]
UPDATE!

OMG after all this time I remembered one CRUCIAL thing. I had a split DNS config for mesh.domain.com on my local network pointing to its local IP. This meant that the agents were expecting CloudfFlare's proxied address and their cert but getting my local IP and the Lets Encrypt cert. I have this set up for WAN only mode and that is why.

In summary if anyone else gets this issue make sure CloudFlare's SSL setting is set to strict and you are able to ping the subdomain and get the WAN address of Cloudflare's proxy. This problem is now resolved! Thank GOD.

Might also be a good idea to re-download the latest agent, uninstall from trouble endpoints and when you install again it should check in immediately.
